src/app/themes.tsx

```diff
@@ -10,53 +10,121 @@ import {
 } from "@mui/material";
 
-type CustomTheme = {
-    foregroundColor: string;
-    backgroundColor: string;
-    primaryForegroundColor: string;
-    primaryBackgroundColor: string;
-    secondaryForegroundColor: string;
-    secondaryBackgroundColor: string;
-    accentBackgroundColor: string;
-    accentForegroundColor: string;
-    surfaceForegroundColor: string;
-    surfaceBackgroundColor: string;
-    errorColor: string;
+export type standardPalette = {
+    background: string[];
+    foreground: string[];
+    largeForeground: string[];
 };
 
-const lightTheme: CustomTheme = {
-    foregroundColor: "#000000",
-    backgroundColor: "#eeeeee",
-    primaryForegroundColor: "#000000",
-    primaryBackgroundColor: "#63a036",
-    secondaryForegroundColor: "#ffffff",
-    secondaryBackgroundColor: "#066940",
-    accentBackgroundColor: "#609fd3",
-    accentForegroundColor: "#000000",
-    surfaceForegroundColor: "#000000",
-    surfaceBackgroundColor: "#ffffff",
-    errorColor: "#ff624e",
+export type rainbowPalette = {
+    red: standardPalette;
+    orange: standardPalette;
+    yellow: standardPalette;
+    lightGreen: standardPalette;
+    darkGreen: standardPalette;
+    blue: standardPalette;
+    purple: standardPalette;
+    grey: standardPalette;
+    brown: standardPalette;
 };
 
-const darkTheme: CustomTheme = {
-    foregroundColor: "#ffffff",
-    backgroundColor: "#1a1a1a",
-    primaryForegroundColor: "#ffffff",
-    primaryBackgroundColor: "#066940",
-    secondaryForegroundColor: "#000000",
-    secondaryBackgroundColor: "#63a036",
-    accentBackgroundColor: "#1b629c",
-    accentForegroundColor: "#ffffff",
-    surfaceForegroundColor: "#ffffff",
-    surfaceBackgroundColor: "#161414",
-    errorColor: "#ff624e",
+export type customTheme = {
+    light: boolean;
+    main: standardPalette & { border: string };
+    primary: standardPalette;
+    error: string;
+    shadow: string;
+    rainbow: rainbowPalette;
+};
+
+export const lightTheme: customTheme = {
+    light: true,
+    main: {
+        background: ["#fdfdfd", "#eeeeee", "#eeeeee", "#eeeeee"],
+        foreground: ["#000000", "#2d2d2d", "#383838", "#4d4d4d"],
+        largeForeground: ["#000000", "#2d2d2d", "#383838", "#4d4d4d"],
+        border: "#d5d5d5",
+    },
+    primary: {
+        background: ["#a3e0b0", "#7fd495", "#02985a", "#05663f", "#154d30"],
+        foreground: ["#000000", "#000000", "#000000", "#f2f2f2", "#f2f2f2"],
+        largeForeground: ["#000000", "#000000", "#f2f2f2", "#f2f2f2", "#f2f2f2"],
+    },
+    error: "#ff624e",
+    shadow: "#e2e2e2",
+    rainbow: {
+        red: {
+            background: ["#fa9189", "#fb685a", "#fc2a2a"],
+            foreground: ["#000000", "#000000", "#000000"],
+            largeForeground: ["#000000", "#000000", "#f2f2f2"],
+        },
+        orange: {
+            background: ["#fcae7c", "#ff9753", "#fa6400"],
+            foreground: ["#000000", "#000000", "#000000"],
```

Comment:
Same comment as before with repeated colours, also do we need say 3 reds or 3 oranges?

Reply:
Ye similarly it is just to not let the use bother with choosing what color to match

Basically the usage is if I am using orange.background[0] I should not have a choice to choose what color to use, but just use orange.foreground[0] for smaller text or either orange.foreground[0] or orange.largeForeground[0] (sometimes accessibility allows larger text to have another color)

So when the user uses the color, they don't have to worry too much about what color text fulfills the accessibility test, they are all out of the box